### PR TITLE
Modify swap constraints, add swap guardian.

### PR DESCRIPTION
**Pull request by gyeiv**
This PR handles [this ticket](https://app.asana.com/0/1199953095960779/1201760072593316/f).

- Changes swap fees to 0.
- Adds a guardian account as signer. Swap actions cannot be taken without this account's signature. This allows us to gate swaps based on additional logic (e.g. prevent swaps past market expiry time).

### CODE REVIEW

**Comment by Juici** [comment]
For a preliminary viewing, there are lots of missing trailing commas which could be fixed with a `cargo fmt`.

**Comment by gyeiv** [comment]
> For a preliminary viewing, there are lots of missing trailing commas which could be fixed with a `cargo fmt`.

If I do format, will probably do in a separate PR (unsure what formatting guidelines their existing repo follows, don't want to introduce extraneous changes in this PR).

**Comment by Juici** [comment]
> > For a preliminary viewing, there are lots of missing trailing commas which could be fixed with a `cargo fmt`.
> 
> If I do format, will probably do in a separate PR (unsure what formatting guidelines their existing repo follows, don't want to introduce extraneous changes in this PR).

From the looks of the workflows, they just run with default rustfmt configs.

https://github.com/Hedgehog-Markets/solana-program-library/blob/master/.github/workflows/pull-request.yml#L23-L44

File: token-swap/program/src/error.rs [hunk 1]
```diff
@@ -102,6 +102,12 @@ pub enum SwapError {
     /// The operation cannot be performed on the given curve
     #[error("The operation cannot be performed on the given curve")]
     UnsupportedCurveOperation,
+    /// Incorrect swap guardian was provided with the operation.
+    #[error("The provided swap guardian is incorrect")]
+    WrongSwapGuardian,
+    /// The swap guardian is not a signer.
+    #[error("No swap guardian signature")]
+    NoSwapGuardianSig
```
Comment (Juici) [inline token-swap/program/src/error.rs]: Do we want to use `Sig` instead of `Signature` here?

File: token-swap/program/src/error.rs [hunk 2]
```diff
@@ -102,6 +102,12 @@ pub enum SwapError {
     /// The operation cannot be performed on the given curve
     #[error("The operation cannot be performed on the given curve")]
     UnsupportedCurveOperation,
+    /// Incorrect swap guardian was provided with the operation.
+    #[error("The provided swap guardian is incorrect")]
+    WrongSwapGuardian,
```
Comment (Juici) [inline token-swap/program/src/error.rs]: This should probably be called `IncorrectSwapGuardian` to match to wording in the other errors.

**Review by Juici — APPROVED** [review]
Other than formatting issues, looks good.

File: token-swap/program/src/error.rs (hunk 1 above)
Comment (gyeiv) [reply token-swap/program/src/error.rs]: Done - changed to Signature

File: token-swap/program/src/error.rs (hunk 2 above)
Comment (gyeiv) [reply token-swap/program/src/error.rs]: Done.

**Comment by tasiov** [comment]
Looks good, though the tests don't pass do they?

**Review by tasiov — APPROVED** [review]

**Comment by gyeiv** [comment]
> Looks good, though the tests don't pass do they?

The tests pass - are they failing for you?

**Pull request merged**